### PR TITLE
Update to new special notes syntax in campaigns

### PR DESCRIPTION
**Pull request by CelticMinstrel**
This is just here to make Travis run a build. As soon as it passes (well, assuming it does), this can be merged. (Preferably not squash-merged.)

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
@gfgtdf - Do I need to do any extra work for `[modify_unit]` to support changing special notes with a nested `[special_note]` tag?

**Comment by gfgtdf** [comment]
> @gfgtdf - Do I need to do any extra work for [modify_unit] to support changing special notes with a nested [special_note] tag?

I actuall thought description is only used in the help and thus is only a property of unit types and not of single units (as there is no was to shop help pages for single units?).

> @gfgtdf - Do I need to do any extra work for [modify_unit] to support changing special notes with a nested [special_note] tag?

not sure depends on whether [special_note] makes sense in [modify_unit], also , what should [modify_unit][special_note] do in the first place, change the first [special_note], add a [special_note] or replace all [special_note]s ?

**Comment by CelticMinstrel** [comment]
> I actually thought description is only used in the help and thus is only a property of unit types and not of single units (as there is no was to shop help pages for single units?).

I thought so too, but UtBS actually customizes descriptions on a per-unit basis for some reason... not sure if those descriptions actually appear anywhere at the moment, but logically it would make sense for them to show in a help page geared towards a specific unit.

> what should [modify_unit][special_note] do in the first place, change the first [special_note], add a [special_note] or replace all [special_note]s ?

This is a good question, to be sure, but I think replacing all would be the easiest, though adding one would also be valid. Changing the first one would probably be confusing.

File: data/campaigns/Heir_To_The_Throne/units/Battle_Princess.cfg [hunk 1]
```diff
@@ -120,7 +122,10 @@
             {ABILITY_INITIATIVE}
         [/abilities]
         usage=fighter
-        description= _ "A noble by birth, the Princess has learnt swordplay with the greatest generals and battle tactics with the greatest sages, making her both a great combatant and a leader. Now battle-hardened and strong of will, she can aid those around her in the art of combat."{SPECIAL_NOTES}+{SPECIAL_NOTES_LEADERSHIP}+{SPECIAL_NOTES_INITIATIVE}+{SPECIAL_NOTES_MAGICAL}
+        description= _ "A noble by birth, the Princess has learnt swordplay with the greatest generals and battle tactics with the greatest sages, making her both a great combatant and a leader. Now battle-hardened and strong of will, she can aid those around her in the art of combat."{SPECIAL_NOTES}
```
Comment (jostephd) [inline data/campaigns/Heir_To_The_Throne/units/Battle_Princess.cfg]: Leftover `{SPECIAL_NOTES}` here, and in Commander.cfg in the same directory. Didn't check if there were other cases.

**Comment by CelticMinstrel** [comment]
Since this is already merged, you can feel free to fix any missed cases yourself. :)

**Comment by jostephd** [comment]
Fixed those two, but didn't look for others.

**Pull request merged**